components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/geo/service/GeoLocationProviderServiceImpl.java

```diff
@@ -530,6 +879,57 @@ public String getProximityAlerts(DeviceIdentifier identifier) throws GeoLocation
         }
     }
 
+    @Override
+    public List<GeoFence> getStationaryAlerts() throws GeoLocationBasedServiceException {
+
+        Registry registry = getGovernanceRegistry();
+        String registryPath = GeoServices.REGISTRY_PATH_FOR_ALERTS +
+                GeoServices.ALERT_TYPE_STATIONARY + "/";
+        Resource resource;
+        try {
+            resource = registry.get(registryPath);
+        } catch (RegistryException e) {
+            log.error("Error while reading the registry path: " + registryPath);
+            return null;
```

Comment:
Shall we throw `GeoLocationBasedServiceException ` wrapping `RegistryException` ?

Reply:
Doing so would throw GeoLocationBasedServiceException when the device location dashboard loads, since resources for the registry paths won't be available until alerts are added for the first time. Therefore, returned an empty Collection instead. c653051